### PR TITLE
WIP usecase-demo UI

### PR DESCRIPTION
**Pull request by ErwanDecoster**
adding back button, rename disclaimer modal, format code with prettier

### CODE REVIEW

File: packages/demo/usecase-demo/src/main.tsx [hunk 1]
```diff
@@ -31,7 +31,7 @@ ReactDOM.createRoot(document.getElementById('root')!).render(
   <React.StrictMode>
     <WagmiConfig config={wagmiConfig}>
       <QueryClientProvider client={queryClient}>
-        <DevModal />
+        <DiscaimerModal />
```
Comment (cedric25) [inline packages/demo/usecase-demo/src/main.tsx:34]: Typo to be fixed, missing one letter it seems.

```suggestion
        <DisclaimerModal />
```

File: packages/demo/usecase-demo/src/modules/DisclaimerModal.tsx [hunk 2]
```diff
@@ -34,32 +33,36 @@ export function DevModal() {
   useEffect(() => {
     setStorageDevMode(isDevMode);
   }, [isDevMode]);
-  
+
   useEffect(() => {
     if (open == false) {
```
Comment (cedric25) [inline packages/demo/usecase-demo/src/modules/DisclaimerModal.tsx:38]: Better to never use this `==` syntax.
I propose here:

```
if (open === false) {
```

File: packages/demo/usecase-demo/src/modules/DisclaimerModal.tsx [hunk 3]
```diff
@@ -34,32 +33,36 @@ export function DevModal() {
...
-      setStorageModalViewed(true);
+      setStorageDiscaimerViewed(true);
     }
   }, [open]);
 
-  if (!isStorageModalViewed) {
+  if (!isStorageDiscaimerViewed) {
```
Comment (cedric25) [inline packages/demo/usecase-demo/src/modules/DisclaimerModal.tsx:43]: We will avoid putting too much code in a big `if` like this.
We prefer to **break early**.

Example here:

if (isStorageDiscaimerViewed) {
  return false;
}

return (
  <Dialog ...

File: packages/demo/usecase-demo/src/routes/_explore/content.$protectedDataAddress.tsx [hunk 4]
```diff
@@ -30,6 +31,10 @@ export function ProtectedDataPreview() {
     address: protectedDataAddress,
   });
 
+  const router = useRouter();
+  const hasPrevionPage = router.history.location.state.key ? true : false;
```
Comment (cedric25) [inline packages/demo/usecase-demo/src/routes/_explore/content.$protectedDataAddress.tsx:35]: Typo to be fixed:

`hasPrevionPage` → `hasPreviousPage`

Comment (cedric25) [reply packages/demo/usecase-demo/src/routes/_explore/content.$protectedDataAddress.tsx:35]: Ternary is probably unnecessary here, we can simplify:

```
const hasPreviousPage = !!router.history.location.state.key;
```

File: packages/demo/usecase-demo/src/main.tsx (hunk 1 above)
Comment (cedric25) [reply packages/demo/usecase-demo/src/main.tsx:34]: I'd put it down below, just before <Toaster />

This `<DisclaimerModal>` doesn't need `WagmiConfig` and `QueryClientProvider` so we can put it outside these providers.

File: packages/demo/usecase-demo/src/modules/DisclaimerModal.tsx [hunk 5]
```diff
@@ -34,32 +33,36 @@ export function DevModal() {
   useEffect(() => {
     setStorageDevMode(isDevMode);
   }, [isDevMode]);
-  
+
   useEffect(() => {
```
Comment (cedric25) [inline packages/demo/usecase-demo/src/modules/DisclaimerModal.tsx:37]: We can probably avoir this useEffect, and setting `StorageDisclaimerViewed` to true in the `onOpenChange` modal callback.

File: packages/demo/usecase-demo/src/modules/DisclaimerModal.tsx [hunk 6]
```diff
@@ -8,22 +10,19 @@ import {
...
     'ContentCreator_devMode',
     { defaultValue: false }
   );
-  const [ isStorageModalViewed, setStorageModalViewed ] = useLocalStorageState(
-    'ContentCreator_modalViewed',
-    { defaultValue: false }
-  );
-  // setStorageModalViewed(false)
+  const [isStorageDiscaimerViewed, setStorageDiscaimerViewed] =
+    useLocalStorageState('ContentCreator_discaimerViewed', {
```
[the remark this answers is not in the document]
Comment (ErwanDecoster) [reply packages/demo/usecase-demo/src/modules/DisclaimerModal.tsx:23]: done

File: packages/demo/usecase-demo/src/routes/_explore/content.$protectedDataAddress.tsx (hunk 4 above)
Comment (ErwanDecoster) [reply packages/demo/usecase-demo/src/routes/_explore/content.$protectedDataAddress.tsx:35]: done

File: packages/demo/usecase-demo/src/modules/DisclaimerModal.tsx (hunk 5 above)
Comment (ErwanDecoster) [reply packages/demo/usecase-demo/src/modules/DisclaimerModal.tsx:37]: done

File: packages/demo/usecase-demo/src/modules/DisclaimerModal.tsx (hunk 3 above)
Comment (ErwanDecoster) [reply packages/demo/usecase-demo/src/modules/DisclaimerModal.tsx:43]: done

File: packages/demo/usecase-demo/src/modules/DisclaimerModal.tsx (hunk 2 above)
Comment (ErwanDecoster) [reply packages/demo/usecase-demo/src/modules/DisclaimerModal.tsx:38]: done

File: packages/demo/usecase-demo/src/main.tsx (hunk 1 above)
Comment (ErwanDecoster) [reply packages/demo/usecase-demo/src/main.tsx:34]: done

**Pull request merged**